### PR TITLE
A trigo pb

### PR DESCRIPTION
**Pull request by jtlap**
a bug on rem180 and suppressing some uneedded options for direct trigo in degrees and pis

### CODE REVIEW

File: include/eve/module/math/regular/acotd.hpp [hunk 1]
```diff
@@ -68,8 +68,10 @@ namespace eve
 //!  @godbolt{doc/math/acotd.cpp}
 //!  @}
 //================================================================================================
-
   inline constexpr auto acotd = functor<acotd_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/acotd.hpp:72]: double comments

File: include/eve/module/math/regular/acsc.hpp [hunk 2]
```diff
@@ -75,9 +75,11 @@ namespace eve
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/acsc.cpp}
 //!  @}
+//================================================================================================
+  inline constexpr auto acsc = functor<acsc_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/acsc.hpp:80]: why not the same length of === ?

File: include/eve/module/math/regular/acscd.hpp [hunk 3]
```diff
@@ -70,9 +70,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/acscd.cpp}
+//================================================================================================
+  inline constexpr auto acscd = functor<acscd_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/acscd.hpp:75]: why not the same length of === ?

File: include/eve/module/math/regular/agd.hpp [hunk 4]
```diff
@@ -74,9 +74,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/agd.cpp}
+//================================================================================================
+  inline constexpr auto agd = functor<agd_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/agd.hpp:79]: why not the same length of === ?

File: include/eve/module/math/regular/asec.hpp [hunk 5]
```diff
@@ -77,9 +77,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/asec.cpp}
+//================================================================================================
+  inline constexpr auto asec = functor<asec_t>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/asec.hpp:81]: why not the same length of === ?

File: include/eve/module/math/regular/asecd.hpp [hunk 6]
```diff
@@ -74,9 +74,11 @@ struct asecd_t : elementwise_callable<asecd_t, Options, raw_option>
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/asecd.cpp}
 //!  @}
+//================================================================================================
+  inline constexpr auto asecd = functor<asecd_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/asecd.hpp:79]: Double =====

File: include/eve/module/math/regular/asech.hpp [hunk 7]
```diff
@@ -69,9 +69,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/asech.cpp}
+//================================================================================================
+  inline constexpr auto asech = functor<asech_t>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/asech.hpp:73]: why not the same length of === ?

File: include/eve/module/math/regular/asin.hpp [hunk 8]
```diff
@@ -79,9 +79,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/asin.cpp}
+//================================================================================================
+  inline constexpr auto asin = functor<asin_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/asin.hpp:84]: why not the same length of === ?

File: include/eve/module/math/regular/atan.hpp [hunk 9]
```diff
@@ -75,9 +75,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/atan.cpp}
+//================================================================================================
+ inline constexpr auto atan = functor<atan_t>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/atan.hpp:79]: why not the same length of === ?

File: include/eve/module/math/regular/atand.hpp [hunk 10]
```diff
@@ -71,8 +71,10 @@ struct atand_t : elementwise_callable<atand_t, Options>
 //!  @godbolt{doc/math/atand.cpp}
 //!  @}
 //================================================================================================
-
   inline constexpr auto atand = functor<atand_t>;
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/atand.hpp:75]: double //===========

File: include/eve/module/math/regular/atanh.hpp [hunk 11]
```diff
@@ -75,9 +75,11 @@ namespace eve
 //!
 //!  @groupheader{Example}
 //!  @godbolt{doc/math/atanh.cpp}
+//================================================================================================
+  inline constexpr auto atanh = functor<atanh_t>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/atanh.hpp:79]: why not the same length of === ?

File: include/eve/module/math/regular/cbrt.hpp [hunk 12]
```diff
@@ -67,9 +67,11 @@ namespace eve
 //!   *  [Wikipedia](https://en.wikipedia.org/wiki/Cube_root)
 //!   *  [Wolfram MathWorld](https://mathworld.wolfram.com/CubeRoot.html)
 //!  @}
+//================================================================================================
+  inline constexpr auto cbrt = functor<cbrt_t>;
```
Comment (jfalcou) [inline include/eve/module/math/regular/cbrt.hpp:71]: double //===============

File: include/eve/module/math/regular/cospi.hpp [hunk 13]
```diff
@@ -19,8 +19,7 @@
 namespace eve
 {
   template<typename Options>
-  struct cospi_t : elementwise_callable<cospi_t, Options, quarter_circle_option, half_circle_option,
-                                        full_circle_option, medium_option, big_option>
+  struct cospi_t : elementwise_callable<cospi_t, Options, quarter_circle_option>
```
Comment (jfalcou) [inline include/eve/module/math/regular/cospi.hpp:22]: cosd accepts big_option. Why not this one ? or is it an error on cosd ?

File: include/eve/module/math/regular/cospi.hpp [hunk 14]
```diff
@@ -76,24 +73,28 @@ namespace eve
...
-//!        2. assumes that the inputs elements  belong to \f$[-1/2,1/2]\f$ and return NaN outside.
-//!        3. assumes that the inputs elements  belong to \f$[-1,1]\f$ and return NaN outside.
+//!    3. Assumes that the inputs elements  belong to \f$[-1/4,1/4]\f$ and return NaN outside.
 //!
 //!  @groupheader{Example}
+//================================================================================================
+ inline constexpr auto cospi = functor<cospi_t>;
+//================================================================================================
 //!  @godbolt{doc/math/cospi.cpp}
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/cospi.hpp:83]: Double //==

File: include/eve/module/math/regular/tanpi.hpp [hunk 15]
```diff
@@ -74,26 +71,31 @@ namespace eve
...
-//!        2. assumes that the inputs elements  belong to \f$[-1/2,1/2]\f$ and return NaN outside.
-//!        3. assumes that the inputs elements  belong to \f$[-1,1]\f$ and return NaN outside.
+//!    3. Assumes that the inputs elements  belong to \f$[-1/4,1/4]\f$ and return NaN outside.
 //!
 //!  @groupheader{Example}
+//================================================================================================
+  inline constexpr auto tanpi = functor<tanpi_t>;
+//================================================================================================
 //!  @godbolt{doc/math/tanpi.cpp}
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/tanpi.hpp:81]: double //===============

File: include/eve/module/math/regular/tand.hpp [hunk 16]
```diff
@@ -73,26 +70,31 @@ namespace eve
...
-//!        2. assumes that the inputs elements  belong to \f$[-90,  90]\f$ and return NaN outside.
-//!        3. assumes that the inputs elements  belong to \f$[-180, 180]\f$ and return NaN outside.
+//!    3. Assumes that the inputs elements  belong to \f$[-45,45]\f$ and return NaN outside.
 //!
 //!  @groupheader{Example}
+//================================================================================================
+  inline constexpr auto tand = functor<tand_t>;
+//================================================================================================
 //!  @godbolt{doc/math/tand.cpp}
+//================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/tand.hpp:80]: DOuble //===========

**Review by jfalcou — CHANGES_REQUESTED** [review]
Lot of bad placemetn of //====

File: include/eve/module/math/regular/tand.hpp (hunk 16 above)
Comment (jtlap) [reply include/eve/module/math/regular/tand.hpp:77]: all done

**Comment by jtlap** [comment]
J'ai tout passé en revue et corrigé

File: include/eve/module/math/regular/rem_pio2.hpp [hunk 17]
```diff
@@ -1,18 +0,0 @@
-//==================================================================================================
```
Comment (jfalcou) [inline include/eve/module/math/regular/rem_pio2.hpp:1]: To delete ???

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/math/regular/rem_pio2.hpp (hunk 17 above)
Comment (jtlap) [reply include/eve/module/math/regular/rem_pio2.hpp]: non je l'ai rétabli. just merge

**Review by jfalcou — APPROVED** [review]

**Pull request merged**